compconfig/compconfig/functions.py

```diff
@@ -0,0 +1,148 @@
+
+
+def run_model(meta_param_dict, adjustment):
+    '''
+    Initializes classes from OG-USA that compute the model under
+    different policies.  Then calls function get output objects.
+    '''
+    # Create output directory structure
+    ss_dir = os.path.join(output_base, "SS")
+    tpi_dir = os.path.join(output_base, "TPI")
+    dirs = [ss_dir, tpi_dir]
+    for _dir in dirs:
+        try:
+            print("making dir: ", _dir)
+            os.makedirs(_dir)
+        except OSError:
+            pass
+    # Dask parmeters
+    client = None
+    num_workers = 1
+
+    # whether to estimate tax functions from microdata
+    run_micro = False
+
+    # Solve baseline model
+    base_spec = {'start_year': meta_param_dict['year'],
+                 'debt_ratio_ss': 1.5, 'debt_ratio_ss': 2.0,
+                 'r_gov_scale': 1.0, 'r_gov_shift': 0.02,
+                 'zeta_D': [0.4], 'zeta_K': [0.1],
+                 'initial_debt_ratio': 0.78,
+                 'initial_foreign_debt_ratio': 0.4}
+    base_params = Specifications(
+        run_micro=False, output_base=BASELINE_DIR,
+        baseline_dir=BASELINE_DIR, test=False, time_path=False,
+        baseline=True, iit_reform={}, guid='',
+        data=meta_param_dict['data_source'],
+        client=client, num_workers=num_workers)
+    base_params.update_specifications(og_spec)
+    base_params.get_tax_function_parameters(client, run_micro)
+    base_ss = SS.run_SS(base_params, client=client)
+
+    # Solve reform model
+    reform_spec = {'start_year': meta_param_dict['year'],
+                   'debt_ratio_ss': 1.5, 'debt_ratio_ss': 2.0,
+                   'r_gov_scale': 1.0, 'r_gov_shift': 0.02,
+                   'zeta_D': [0.4], 'zeta_K': [0.1],
+                   'initial_debt_ratio': 0.78,
+                   'initial_foreign_debt_ratio': 0.4}.update(adjustment)
+    reform_params = Specifications(
+        run_micro=False, output_base=BASELINE_DIR,
+        baseline_dir=BASELINE_DIR, test=False, time_path=False,
+        baseline=False, iit_reform={}, guid='',
+        data=meta_param_dict['data_source'],
+        client=client, num_workers=num_workers)
+    reform_params.update_specifications(reform_spec)
+    reform_params.get_tax_function_parameters(client, run_micro)
+    reform_ss = SS.run_SS(reform_params, client=client)
+
+    comp_dict = comp_output(base_ss, base_params, reform_ss,
+                            reform_params)
+
+    return comp_dict
+
+
+def comp_output(base_ss, base_params, reform_ss, reform_params,
+                var='nssmat'):
+    '''
+    Function to create output for the COMP platform
+    '''
+    plt = op.ss_profiles(
+        base_ss, base_params, reform_ss=reform_ss,
+        reform_params=reform_params, by_j=True, var=var,
+        plot_title='Labor Supply in Baseline and Reform Policy')
+    comp_dict = {
+        "model_version": ogusa.__version__,
+        "renderable": [
+            {
+              "media_type": "matplotlib",
```

Comment:
@hdoupe I couldn't find an example of using a matplotlib figure.  Can you suggest how this might be done?

Reply:
@jdebacker Is it interactive? If not, you can dump it to a PNG file. I'll add an example to the docs showing how to do that.

Reply:
@hdoupe It is not interactive - I just want a static png file.  Thanks!

Reply:
@jdebacker here's an example for matplotlib results: https://docs.compute.studio/publish/outputs/#pngjpeg